Modules/mathmodule.c

```diff
@@ -3802,7 +3805,14 @@ math_nextafter_impl(PyObject *module, double x, double y)
         return PyFloat_FromDouble(y);
     }
 #endif
-    return PyFloat_FromDouble(nextafter(x, y));
+    if (steps < 1) {
+        PyErr_SetString(PyExc_ValueError, "steps must be >= 1");
+        return NULL;
+    }
+    for (i = 0; i < steps; i++) {
```

Comment:
You can support this without a loop.  See [this pure Python prototype](https://gist.github.com/matthiasgoergens/28e58abd6beaea72bd2e49085c435966) for how that can work.